make.py

```diff
@@ -0,0 +1,249 @@
+                                 'provided, assume it is installed where '
+                                 'Langkit must be installed.')
+install_parser.add_argument('--gnatcoll-bindings-install-dir', type=str,
+                            default=None,
+                            help='Specify where GNATCOLL bindings are '
+                                 'installed. If not provided, assume it is '
+                                 'installed where Langkit must be installed.')
+install_parser.add_argument('--libgpr-install-dir', type=str,
+                            default=None,
+                            help='Specify where libgpr is installed. If not '
+                                 'provided, assume it is installed where '
+                                 'Langkit must be installed.')
+
+
+ROOT_DIR = os.path.dirname(__file__)
+
+if ROOT_DIR != '':
+    os.chdir(ROOT_DIR)
+
+LIBPYTHONLANG_DIR = os.path.join('contrib', 'python')
+
+
+@contextmanager
+def in_dir(path):
+    """
+    Move the working directly to the given path for the span of this context
+    manager.
+
+    :type path: str
+    """
+    old_dir = os.getcwd()
+    os.chdir(path)
+    yield
+    os.chdir(old_dir)
+
+
+def pip_install(setup_dir):
+    """
+    Install the python package located in the given directory.
+
+    :type setup_dir: str
+    :raise CalledProcessError if installation terminates with non-zero status
+    """
+    with in_dir(setup_dir):
+        subprocess.check_call([sys.executable, '-m', 'pip', 'install', '.'])
+
+
+def copy_all_shared_libs(from_dir, destination, *patterns):
+    """
+    Copy all shared libraries from anywhere in a given source tree to a given
+    destination folder.
+
+    Only the libraries that satisfy the given simple patterns are copied.
+    Given patterns must NOT be regular expressions, but simple strings.
+    A library will match if the pattern is a substring of the library's name.
+
+    :type from_dir: str
+    :type destination: str
+    :type patterns: list[str]
+    """
+    for root, _, filenames in os.walk(from_dir):
+        for filename in filenames:
+            if any(n in filename for n in patterns):
+                if any(ext in filename for ext in (".so", ".dll", ".dylib")):
+                    shutil.copy(os.path.join(root, filename), destination)
+
+
+def manage(lang_dir, verb, build_mode, *add_args):
+    """
+    Run the manage.py script from of a Langkit-based language definition.
+
+    Raises CalledProcessError if 'manage.py' terminates with non-zero status.
+
+    :param str verb: 'generate', 'build', 'install', etc.
+    :param str build_mode: 'dev' or 'prod'.
+    :param list[str] add_args: additional arguments to pass
+    """
+    with in_dir(lang_dir):
+        args = [sys.executable, 'manage.py', verb]
+        args.extend(['--build-mode', build_mode])
+        args.extend(add_args)
+
+        if verb in ('build', 'make'):
+            # Always build shared libs with "$ORIGIN" rpaths.
```

Comment:
IIUC, `-R` disables run paths, so this comment is wrong. I suggest instead:
> Do not use run path options
(from `gprbuild --help`)

Reply:
Well, the whole idea of this section is to put `$ORIGIN` as rpath, so first prevent gprbuild from passing its own rpaths using `-R` and then set the `LD_RUN_PATH` environment variable to `$ORIGIN` to make sure the linker uses it.